src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Single.cs

```diff
@@ -0,0 +1,128 @@
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        ~SingleShot()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Invokes the model with the given input data.
+        /// </summary>
+        /// <param name="inTensorsData">The input data to be inferred.</param>
+        /// <returns>TensorsData instance which contains the inferred result.</returns>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="IOException">Thrown when failed to push a input data into source element.</exception>
+        /// <exception cref="TimeoutException">Thrown when failed to get the result from sink element.</exception>
+        /// <since_tizen> 5.5 </since_tizen>
+        public TensorsData Invoke(TensorsData inTensorsData)
+        {
+            TensorsData out_data;
+            IntPtr out_ptr;
+            NNStreamerError ret = NNStreamerError.None;
+
+            ret = Interop.Single.InvokeSingle(_handle, inTensorsData.Handle, out out_ptr);
+            NNStreamer.CheckException(ret, "fail to invoke the single inference engine");
+
+            out_data = new TensorsData(out_ptr);
+            return out_data;
+        }
+
+        private void CreateSingleShot(string modelAbsPath, TensorsInfo inTensorInfo, TensorsInfo outTensorInfo)
+        {
+            NNStreamerError ret = NNStreamerError.None;
+            IntPtr input_info;
+            IntPtr output_info;
+
+            /* Check model path */
+            if (string.IsNullOrEmpty(modelAbsPath))
+                ret = NNStreamerError.InvalidParameter;
+            NNStreamer.CheckException(ret, "model path is invalid: " + modelAbsPath);
+
+            input_info = inTensorInfo.GetTensorsInfoHandle();
+            output_info = outTensorInfo.GetTensorsInfoHandle();
+
+            ret = Interop.Single.OpenSingle(out _handle, modelAbsPath, input_info, output_info, NNFWType.Any, HWType.Any);
+            NNStreamer.CheckException(ret, "fail to open the single inference engine");
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                NNStreamerError ret = NNStreamerError.None;
+                ret = Interop.Single.CloseSingle(_handle);
```

Comment:
`Interop.Single.CloseSingle()` function internally close all handle and release allocated memory space.  IMHO, it would be enough to handle unmanaged object. Should I do more? If so, please let me know more detail.

Reply:
`Interop.Single.CloseSingle()` is enough, but Dispose(true) is only called when developer was call Dispose(), 
if a object was released by GC, Dispose(false) is called in destructor of this class, in this case `Interop.Single.CloseSingle()` was not called